modules/web/src/app/core/services/kubeone-cluster-spec.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Injectable} from '@angular/core';
+import {ExternalCloudSpec, ExternalCluster} from '@shared/entity/external-cluster';
+import {KubeOneCloudSpec, KubeOneClusterSpec} from '@shared/entity/kubeone-cluster';
+import {NodeProvider} from '@shared/model/NodeProviderConstants';
+import _ from 'lodash';
+import {Subject} from 'rxjs';
+
+@Injectable()
+export class KubeOneClusterSpecService {
```

Comment:
I followed the same approach as `ClusterSpecService`/`NodeSpecService` because in `ClusterService` we have API calls and for data storage/propagation we use `ClusterSpecService`.